app/src/main/java/com/sunlightlabs/congress/services/LegislatorService.java

```diff
@@ -45,16 +45,148 @@
+        for (int i=0; i<representatives.size(); i++) {
+            representatives.get(i).state = state;
+            representatives.get(i).chamber = "house";
+        }
+
+        members.addAll(senators);
+        members.addAll(representatives);
+
+        return members;
+    }
+
 	public static Legislator find(String bioguideId) throws CongressException {
-		Map<String,String> params = new HashMap<String,String>();
-		params.put("bioguide_id", bioguideId);
-		params.put("all_legislators", "true");
-		return legislatorFor(Congress.url("legislators", basicFields, params));
+        String[] paths = new String[] {"members", bioguideId};
+		return proPublicaLegislatorFor(ProPublica.url(paths));
 	}
 	
 	/* JSON parsers, also useful for other service endpoints within this package */
 
+	protected static Legislator fromProPublica(JSONObject json) throws JSONException, CongressException {
+        if (json == null)
+            return null;
+
+        Legislator legislator = new Legislator();
+
+        if (!json.isNull("member_id"))
+            legislator.bioguide_id = json.getString("member_id");
+        // on some endpoints, the bioguide ID is in the 'id' field
+        else if (!json.isNull("id"))
+            legislator.bioguide_id = json.getString("id");
+
+        if (!json.isNull("govtrack_id"))
+            legislator.govtrack_id = json.getString("govtrack_id");
+
+        // API may return in_office as a string, but JSONObject should auto-coerce:
+        // https://developer.android.com/reference/org/json/JSONObject.html
+        if (!json.isNull("in_office"))
+            legislator.in_office = json.getBoolean("in_office");
+
+        if (!json.isNull("first_name"))
+            legislator.first_name = json.getString("first_name");
+        if (!json.isNull("last_name"))
+            legislator.last_name = json.getString("last_name");
+
+        if (!json.isNull("gender"))
+            legislator.gender = json.getString("gender");
+
+        if (!json.isNull("url"))
+            legislator.website = json.getString("url");
+
+        if (!json.isNull("youtube_account")) {
+            String youtube = json.getString("youtube_account");
+            if (!youtube.isEmpty()) legislator.youtube_id = youtube;
+        }
+        if (!json.isNull("twitter_account")) {
+            String twitter = json.getString("twitter_account");
+            if (!twitter.isEmpty()) legislator.twitter_id = twitter;
+        }
+        if (!json.isNull("facebook_account")) {
+            String facebook = json.getString("facebook_account");
+            if (!facebook.isEmpty()) legislator.facebook_id = facebook;
+        }
+
+        // Some fields come from the legislator's current role.
+        // If we have a roles array, use it for some data.
+        // Otherwise, see if we can scrounge up data on the main object.
+        // TODO: a Role object on Legislator?
```

Comment:
We have two separate parsing tracks, a more complete one if it's using the member lookup endpoint, and a smaller one if it's parsing a list of members by state/district.